addons/sale_project/models/sale_order_line.py

```diff
@@ -60,7 +60,7 @@ def default_get(self, fields):
                     }
 
                 if not sale_order:
-                    company_id = self.env.context.get('defaut_company_id', False)
+                    company_id = self.env.context.get('defaut_company_id', self.env.company.id)
```

Comment:
I think you can leave the `False`, there is however a typo in the context key :sweat_smile: 
```suggestion
                    company_id = self.env.context.get('default_company_id', False)
```

Reply:
I don't know why it wouldn't work, strange :confused: 
I will have a look tomorrow

Reply:
After investigating, it turns out that it keeps the `default_company_id` context key in the create, which has a value of `False` when the project has no company. Therefore, we will indeed need to keep the default value you added. The code can however be simplified a bit, because then we know that we will always provide the create with a company_id. I would suggest something like this:
```py
            if 'order_id' in fields and not res.get('order_id'):
                assert (partner_id := self.env.context.get('default_partner_id'))
                project_id = self.env.context.get('link_to_project')
                sale_order = None
                so_create_values = {
                    'partner_id': partner_id,
                    'company_id': company_id = self.env.context.get('defaut_company_id', self.env.company.id),
                }
                if project_id:
                    try:
                        project_so = self.env['project.project'].browse(project_id).sale_order_id
                        project_so.check_access_rule('write')
                        sale_order = project_so
                    except AccessError:
                        pass

                    if not sale_order:
                        so_create_values['project_ids'] = [Command.link(project_id)]

                if not sale_order:
                    sale_order = self.env['sale.order'].create(so_create_values)
                    sale_order.action_confirm()
                default_values['order_id'] = sale_order.id
```
(I didn't test this code)

Reply:
(Also, don't forget to fix the typo in the context key)